src/Relations/MorphToMany.php

```diff
@@ -0,0 +1,461 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Relations;
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\MorphToMany as EloquentMorphToMany;
+use Illuminate\Support\Arr;
+
+use function array_diff;
+use function array_key_exists;
+use function array_keys;
+use function array_map;
+use function array_merge;
+use function array_reduce;
+use function array_values;
+use function count;
+use function is_array;
+use function is_numeric;
+use function is_string;
+use function str_contains;
+
+class MorphToMany extends EloquentMorphToMany
+{
+    /**
+     * Get the key for comparing against the parent key in "has" query.
+     *
+     * @return string
+     */
+    public function getHasCompareKey()
+    {
+        return $this->getForeignKey();
+    }
+
+    /** @inheritdoc */
+    public function getRelationExistenceQuery(Builder $query, Builder $parentQuery, $columns = ['*'])
+    {
+        return $query;
+    }
+
+    /** @inheritdoc */
+    protected function hydratePivotRelation(array $models)
+    {
+        // Do nothing.
+    }
+
+    /**
+     * Set the select clause for the relation query.
+     *
+     * @return array
+     */
+    protected function getSelectColumns(array $columns = ['*'])
+    {
+        return $columns;
+    }
+
+    /** @inheritdoc */
+    protected function shouldSelect(array $columns = ['*'])
+    {
+        return $columns;
+    }
+
+    /** @inheritdoc */
+    public function addConstraints()
+    {
+        if (static::$constraints) {
+            $this->setWhere();
+        }
+    }
+
+    /** @inheritdoc */
+    public function addEagerConstraints(array $models)
+    {
+        // To load relation's data, we act normally on MorphToMany relation,
+        // But on MorphedByMany relation, we collect related ids from pivot column
+        // and add to a whereIn condition
+        if ($this->getInverse()) {
+            $ids = $this->getKeys($models, $this->table);
+            $ids = $this->extractIds($ids[0] ?? []);
+            $this->query->whereIn($this->relatedKey, $ids);
+        } else {
+            parent::addEagerConstraints($models);
+
+            $this->query->where($this->qualifyPivotColumn($this->morphType), $this->morphClass);
```

Comment:
The `$this->table` is the representer of the pivot column's name, not the collection name.
Let's assume we have the `labelleds` pivot column. the `$this->morphType` returns `labelled_type`. So, `qualifyPivotColumn` method just prefixes the  `$this->morphType` with pivot columns. the result of `qualifyPivotColumn` is `labelleds.labelled_type`.
If I neutralize this method, it will throw errors.